src/store.js

```diff
@@ -141,34 +141,18 @@ class IndexedFormula extends Formula { // IN future - allow pass array of statem
         // console.log('ds before substitute: ' + ds)
         if (binding) ds = ds.substitute(binding)
         // console.log('applyPatch: delete: ' + ds)
-        ds = ds.statements
-        var bad = []
-        var ds2 = ds.map(function (st) { // Find the actual statemnts in the store
+        ds.statements.forEach(function (st) { // Find the actual statemnts in the store
           var sts = targetKB.statementsMatching(st.subject, st.predicate, st.object, target)
-          if (sts.length === 0) {
-            // log.info("NOT FOUND deletable " + st)
-            bad.push(st)
-            return null
-          } else {
-            // log.info("Found deletable " + st)
-            return sts[0]
+          if (sts.length > 0) {
+            targetKB.remove(sts[0])
```

Comment:
@timbl Unsure why we are only removing the first matching statement—I think it should be all. But this follows the original code.